### DISCUSSION TITLE
How do I specify the old fontSize and constraint parameters of FormattedText in Avalonia 11.0.9?

### DISCUSSION THREAD

**Discussion by xgirones** [AvaloniaUI/Avalonia#14659] [Q&A]
I have the following piece of code I used to draw a single character with size specified in pixels that worked as expected with Avalonia 0.10.21. 

```
    FormattedText glyph =
        new FormattedText(text: e.glyph.ToString(),
                          typeface: new Typeface("Tahoma"),
                          fontSize: (double)e.bbox.Height,
                          textAlignment: TextAlignment.Center,
                          textWrapping: TextWrapping.NoWrap,
                          constraint: new Size(e.bbox.Width, e.bbox.Height));
                          
    ctx.DrawText(foreground: new SolidColorBrush(c3),
                 origin: new Point((double)e.bbox.Left + (double)e.bbox.Width / 2.0,
                                   (double)e.bbox.Bottom + (double)e.bbox.Height * 1.2f / 2.0f),
                 glyph);          
```

I am now attempting to migrate to Avalonia 11.0.9, and I observe that the constructor of Formatted text takes different parameters. How can I accomplish the same results using the new version?

**Comment by maxkatz6** [comment]
This API is now mirroring WPF, and you probably need `TextLayout` instead.

**Reply by xgirones** [reply]
Thanks. What would be the equivalent of the 'constraint' parameter in TextLayout?

**Reply by xgirones** [reply]
Any advice? How can I render an individual text character such that it stretches in both x and y dimensions to fully span a rectangle defined in pixel units?

**Reply by Gillibald** [reply]
https://learn.microsoft.com/en-us/dotnet/api/system.windows.media.formattedtext?view=windowsdesktop-8.0 we share the same API

**Reply by Gillibald** [reply]
Stretching glyphs to a bounding box isn't possible directly but you can create a geometry from a FormattedText and use that.

**Reply by xgirones** [reply]
Thanks. I will try employing a geometry as you suggest.
